elementary/files
# Fix optical drive change

Make some changes to GOFDirectoryAsync creation, initialization and reloading to fix reference counting and cause the directory to be uncached/destroyed when no longer in use. 

see issue #85.

If the directory is not destroyed and a different remote folder with the same url is loaded, the contents of the cached folder will be shown instead.  These changes fix that issue.

* Do not cache directory until it has initialised.
* Only add back normal ref when removing toggle ref.
* Do not reconnect directory loading handlers on clearing in AbstractDirectoryView, unless reloading.

TO TEST:
Insert, mount and display a optical disk with few tracks.
Eject the disk
Insert a mount another optical disk with more tracks.

In trunk, the View shows the contents of the first disk again.
In branch the View shows the contents of the new disk, as expected.
